taf/validation.py

```diff
@@ -197,5 +198,12 @@ def _compare_commit_with_targets_metadata(
 def _get_unchanged_targets_metadata(auth_repo, updated_role):
     taf_repo = Repository(auth_repo.path)
     all_roles = taf_repo.get_all_targets_roles()
-    all_roles.remove(updated_role)
+    yk_inserted = is_inserted()
+    updated_roles = [updated_role]
+    if yk_inserted:
```

Comment:
I think that this might be coupling our specific needs to closely with this validation (which is supposed to be as generic as possible while still covering out requirements). Can this function receive a list of updated roles instead, so that the YubiKey logic is fully in our closed source code?

Reply:
Thanks, list of updated roles are now in #246